IOPool/Input/test/testSchemaEvolution.sh

```diff
@@ -0,0 +1,93 @@
+#     file (testSchemaEvolution.sh), then you will need to manually
+#     merge in the commit that added the file. Most of the files are
+#     completely new files, although BuildFile.xml, classes.h, and
+#     classes_def.xml already existed may require manually merging a
+#     few lines of code.
+#
+#     Manually edit 2 files.
+#
+#     1. Modify the following line in the file:
+#     DataFormats/TestObjects/interface/SchemaEvolutionTestObjects.h
+#     //#define DataFormats_TestObjects_USE_OLD
+#     To generate an input file with the old format remove the "//"
+#     at the beginning so the macro is defined.
+#
+#     2. In file: DataFormats/TestObjects/src/classes_def.xml
+#     There is a section of code starting with SchemaEvolutionChangeOrder
+#     continuing to SchemaEvolutionMapToUnorderedMap. This section appears
+#     twice. The first section defines old formats that should be used
+#     to generate old format data files (These are all ClassVersion= "3")
+#     When reading, use the new version (the section with ClassVersion="4"
+#     which should be the one enabled in the CMSSW repository).
+#     Of these two sections, exactly one should be commented out.
+#     To generate the input data files, you will need to manually
+#     enable the ClassVersion 3 section.
+#
+#     Then rebuild the working area.
+#
+#     Then run the configuration IOPool/Input/test/SchemaEvolution_create_test_file_cfg.py
+#     This will generate an output file named SchemaEvolutionTest.root.
+#     Rename this file appropriately to include the release used and use
+#     it as an input for this test by adding additional cases below.
+#     The new data file will need to added to the cms-data repository
+#     named IOPool-Input.
+
+file=SchemaEvolutionTestOLD13_2_3.root
+inputfile=$(edmFileInPath IOPool/Input/data/$file) || die "Failure edmFileInPath IOPool/Input/data/$file" $?
+cmsRun ${LOCAL_TEST_DIR}/SchemaEvolution_test_read_cfg.py -- --inputFile "$inputfile" || die "Failed to read old file $file" $?
+
+file=SchemaEvolutionTestOLD13_0_0.root
+inputfile=$(edmFileInPath IOPool/Input/data/$file) || die "Failure edmFileInPath IOPool/Input/data/$file" $?
+# These fail because there was a bug in the version of ROOT associated with CMSSW_13_0_0
+# The bug caused StreamerInfo objects to missing from the ROOT file. In this case,
+# schema evolution fails and also the testForStreamerInfo.C script will find
+# missing StreamerInfo objects.
+# Lets keep this code around because it may be useful if we need to
+# do additional work related to data files written using an executable
+# built from code having the bug.
+#cmsRun ${LOCAL_TEST_DIR}/SchemaEvolution_test_read_cfg.py -- --inputFile "$inputfile" || die "Failed to read old file $file" $?
+#root.exe -b -l -q file:$inputfile "${LOCAL_TEST_DIR}/testForStreamerInfo.C(gFile)" | sort -u | grep Missing > testForStreamerInfo2.log
+#grep "Missing" testForStreamerInfo2.log && die "Missing nested streamer info" 1
```

Comment:
Do I understand correctly that uncommenting these three lines would make the test fail (and thus demonstrates if we would have had these tests before, we would have caught the schema evolution problem earlier)?

Reply:
Yes. The VectorVectorElementNonSplit case would fail.

If the exact same problem recurs, then the test on line 23 above would detect it.
```
grep "Missing" testForStreamerInfo1.log && die "Missing nested streamer info" 1
```

Looking at this some more I think I could add a couple more test criteria at that point that would be useful. I'll do that tomorrow.